mantidproject/scriptrepository
# Straight-forward changes made by 2to3.py means now 2/3 compatible

Subset of files in repository for which the changes by 2to3.py script are judged to be minor and low-risk (mostly print functions, range -> list(range) and similar for dict.keys() etc. and some maps have been replaced with list comprehension). In addition I have added a line to import print_function (removed by 2to3.py) to maintain python 2 compatibility. I did this with a script which adds the line to the top of the file (after the first block of comments if present). Note in some cases there are blank lines between comments at the top of the file and so it might not be added where the other imports are.

A code-check has been performed to check the files will run in python 2 and 3 - but this has not been tested.

To test:
(1) Review the code changes
(2)  Test a subset of the scripts to confirm they run in python 2 and 3.
 
In some cases the script may depend on another script for which I haven't committed the changes from python 2 to 3 as they are less trivial/I am not sure they will work. A spreadsheet with a list of files not committed and the reason are attached.

[Python3_mantidScriptRepo.xlsx](https://github.com/mantidproject/scriptrepository/files/3851663/Python3_mantidScriptRepo.xlsx)
